accepted/0000-workspaces-config-management.md

```diff
@@ -0,0 +1,98 @@
+# npm workspaces: Config management
+
+## Summary
+
+Series of subcommands to allow managing what paths/folders should be handled as **workspaces**.
+
+## Motivation
+
+Minimize manual editing of the `package.json` file while providing more resilient and automated workflows for defining **workspaces** within your project.
+
+## Detailed Explanation
+
+- Add positional `workspaces` argument e.g: `lib/workspaces.js` that provides subcommands to enable workflows for **workspaces*** config management.
+
+## Rationale and Alternatives
+
+- The obvious alternative is to choose to not provide subcommands to handle **workspaces** config management.
+- Others? TBD
+
+## Implementation
+
+```
+# Given a structure:
+.
+├── package.json { "name": "foo" }
+└── packages
+    ├── dep-a
+    │   └── package.json { "name": "dep-a", "version": "1.0.0" }
+    └── dep-b
+        └── package.json { "name": "dep-b", "version": "1.3.1" }
+
+$ npm workspaces ls
+
+$ npm workspaces add ./packages/dep-a
```

Comment:
Reading `npm workspaces add <package>` I would not expect it to mutate the root config. I would expect it to mirror `npm workspaces install tap` in #117 since `npm add/rm` are already aliases to install/remove dependencies.

I think `npm workspaces <subcommand>` can either be a command that operates on `package.json#workspaces` or every `<workspaces>/package.json` mixing and matching is confusing.

One thing we did in [Bolt](https://github.com/boltpkg/bolt) was have a `bolt project/p` command that allowed you to run commands on the root package containing all the workspaces.

Alternatives:

- `npm workspaces-config add/rm`
- `npm project workspaces add/rm`
- `npm workspaces add-workspace/rm-workspace`